hatasmota/entity.py

```diff
@@ -98,27 +104,65 @@ def __init__(self, **kwds: Any):
         self._on_availability_callback: Callable[
             [bool], Coroutine[Any, Any, None]
         ] | None = None
+        self.uses_deep_sleep = False
+        self.deep_sleep_interval = None
+        self.last_up=0
+        self.available=None
         super().__init__(**kwds)
 
     def get_availability_topics(self) -> dict:
         """Return MQTT topics to subscribe to for availability state."""
 
         async def availability_message_received(msg: ReceiveMessage) -> None:
             """Handle a new received MQTT availability message."""
+            last_up_retain=self.last_up
             if msg.payload == self._cfg.availability_online:
+                if self.last_up and self.deep_sleep_interval is None:
+                    self.deep_sleep_interval = int(time.time() - self.last_up)
                 await self.poll_status()
+            else:
+                self.last_up=time.time()
+            available=self.available
+            if msg.payload == self._cfg.availability_online:
+                available=True
+            if msg.payload == self._cfg.availability_offline and not(self.uses_deep_sleep):
+                if not(self.uses_deep_sleep):
+                    available=False
+                else:
+                    _LOGGER.debug("inhibit deep sleep %s", msg.topic)
```

Comment:
Shouldn't we start a timer here, which if it expires sets availability to `False` and calls the availability callback?

You can use this timer class: https://github.com/emontnemery/hatasmota/blob/8a19e8da7b4e398cd5dfb03f8ff1e3a541fab1e3/hatasmota/mqtt.py#L17-L33